custodial-copy-re-indexer/src/test/scala/uk/gov/nationalarchives/reindexer/ReIndexerSpec.scala

```diff
@@ -0,0 +1,140 @@
+import java.util.UUID
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.xpath.{XPathExpression, XPathFactory}
+
+class ReIndexerSpec extends AnyFlatSpec with EitherValues:
+  val uuids: List[UUID] = List(UUID.randomUUID, UUID.randomUUID, UUID.randomUUID)
+  val xpath: XPathExpression = XPathFactory.newInstance.newXPath.compile("//B")
+  val appConfig: Config = Config("", "", "")
+
+  def expectedXPath(xPathExpression: XPathExpression, xpathToCheck: String): Boolean = {
+    val factory = DocumentBuilderFactory.newInstance
+    factory.setNamespaceAware(true)
+    val doc = factory.newDocumentBuilder.parse(new ByteArrayInputStream("<A><B>test</B></A>".getBytes))
+    xPathExpression.evaluate(doc) == XPathFactory.newInstance.newXPath.compile(xpathToCheck).evaluate(doc)
+  }
+
+  "reIndex" should "pass the correct values to the OCFL and Database methods" in {
+    val appConfig = Config("", "", "")
+    given Database[IO] = new Database[IO]:
+      override def getIds: fs2.Stream[IO, UUID] = fs2.Stream.emits(uuids)
+
+      override def write(columnName: String)(updates: Chunk[Configuration.ReIndexUpdate]): IO[Int] = {
+        updates.toList.sortBy(_.id) should equal(uuids.sorted.map(uuid => IoUpdate(uuid, s"$uuid-IO")))
+        IO(updates.size)
+      }
+
+    given Ocfl[IO] = new Ocfl[IO]:
+      override def readValue(ioId: UUID, fileType: FileType, xpath: XPathExpression)(using
+          configuration: Configuration
+      ): IO[List[Configuration.ReIndexUpdate]] = {
+        uuids.contains(ioId) should equal(true)
+        fileType should equal(FileType.IO)
+        expectedXPath(xpath, "//B") should be(true)
+        IO(IoUpdate(ioId, s"$ioId-$fileType") :: Nil)
+      }
+
+    ReIndexer[IO].reIndex(ReIndexArgs(FileType.IO, "", xpath)).unsafeRunSync()
+  }
+
+  "reIndex" should "chunk the updates to the write method" in {
+    val appConfig = Config("", "", "")
+
+    given Database[IO] = new Database[IO]:
+      override def getIds: fs2.Stream[IO, UUID] = fs2.Stream.emits(List.fill(1000)(UUID.randomUUID))
+
+      override def write(columnName: String)(updates: Chunk[Configuration.ReIndexUpdate]): IO[Int] = {
+        updates.toList.size should equal(100)
+        IO(updates.size)
+      }
+
+    given Ocfl[IO] = new Ocfl[IO]:
+      override def readValue(ioId: UUID, fileType: FileType, xpath: XPathExpression)(using
+          configuration: Configuration
+      ): IO[List[Configuration.ReIndexUpdate]] = {
+        IO(IoUpdate(ioId, s"$ioId-$fileType") :: Nil)
+      }
+
+    ReIndexer[IO].reIndex(ReIndexArgs(FileType.IO, "", xpath)).unsafeRunSync()
+  }
+
+  "reIndex" should "not call the OCFL repo or the write method if there are no ids returned" in {
+    given Database[IO] = new Database[IO]:
+      override def getIds: fs2.Stream[IO, UUID] = fs2.Stream.emits(Nil)
+
+      override def write(columnName: String)(updates: Chunk[Configuration.ReIndexUpdate]): IO[Int] = {
+        false should be(true)
```

Comment:
I'm assuming you did this so that if this method is called, then the test would fail? Do you think it's worth just raising an error with a message like "The 'write' method should not have been called since no ids were returned"? Same question for the `readValue` method?